### PR TITLE
[lmdb] Remove 'd' suffix from debug libraries

### PR DESCRIPTION
**Pull request by bold84**
**Describe the pull request**

- #### What does your PR fix?
  'd' won't be appended to library files on *nix platforms anymore

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JackBoosY** [comment]
Can you please explain why you made this changes?

**Comment by bold84** [comment]
> Can you please explain why you made this changes?

Building zeroc-ice:arm64-osx-dynamic fails because the linker can't find "-llmbd" while linking IceDB library.

File: ports/lmdb/cmake/CMakeLists.txt [hunk 1]
```diff
@@ -27,8 +27,9 @@ endif()
 
 set(SRCS lmdb mdb.c lmdb.h midl.c midl.h )
 add_library(lmdb ${SRCS})
-set_target_properties(lmdb PROPERTIES DEBUG_POSTFIX d)
-if (WIN32)
+
+if (VCPKG_TARGET_IS_WINDOWS)
```
Comment (dg0yt) [inline ports/lmdb/cmake/CMakeLists.txt:31]: `VCPKG_TARGET_IS_WINDOWS` works in vcpkg portfiles (script mode), but not in the actual projects (project mode).

**Review by dg0yt — COMMENTED** [review]
Maybe the debug postfix shouldn't be added at all? Is there an upstream standard?

File: ports/lmdb/cmake/CMakeLists.txt (hunk 1 above)
Comment (bold84) [reply ports/lmdb/cmake/CMakeLists.txt:31]: Oh... good one! True... It's actually the project file. 😂

I'll revert it back to WIN32.

**Comment by JackBoosY** [comment]
> > Can you please explain why you made this changes?
> 
> Building zeroc-ice:arm64-osx-dynamic fails because the linker can't find "-llmbd" while linking IceDB library.

So I assume we don't need to add suffix `d` in any triplet, right?

**Comment by bold84** [comment]
> > > Can you please explain why you made this changes?
> 
> > 
> 
> > Building zeroc-ice:arm64-osx-dynamic fails because the linker can't find "-llmbd" while linking IceDB library.
> 
> 
> 
> So I assume we don't need to add suffix `d` in any triplet, right?

I will test without it within the next hour.

**Comment by bold84** [comment]
@JackBoosY Done

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Confirming that this was present in the original PR https://github.com/microsoft/vcpkg/commit/1f63deb204b555046d525d6ecb3b265e2774f4ba and not something where we're just flipping the bit on which bugs we trigger.

Considering we're providing the entire build system as a replacement seems reasonable to me that DEBUG_POSTFIX should default to empty string.

Thanks!

**Pull request merged**